src/container/srv_container.c

```diff
@@ -3147,9 +3149,19 @@ close_iter_cb(daos_handle_t ih, d_iov_t *key, d_iov_t *val, void *varg)
 		return rc;
 
 	uuid_copy(buf->rb_recs[buf->rb_nrecs].tcr_hdl, key->iov_buf);
-	buf->rb_recs[buf->rb_nrecs].tcr_hce = hdl->ch_hce;
+	buf->rb_recs[buf->rb_nrecs].tcr_hce = cont_hdl->ch_hce;
 	buf->rb_nrecs++;
 
+	pool_hdl = ds_pool_hdl_lookup(cont_hdl->ch_pool_hdl);
```

Comment:
Not 100% sure on whether it's a good idea, but an observation is that the caller (ds_cont_close_by_pool_hdls()) has just looked up the container service (struct cont_svc) that already has (unless I'm not thinking right) the struct ds_pool *cs_pool - needed here for the metrics update. @liw what do you think, would it be too much of a hack for that caller code to provide the struct ds_pool * in the struct close_iter_arg used here, so this function doesn't have to do the pool handle lookup?